### PR TITLE
Upgrade react-cra example to use ts_project, jest_test & js_run_devserver

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

**Comment by gregmagolan** [comment]
@jbedard The BUILD fine under `src` follows the conventions we are planning for BUILD file generation (minus the eslint_test target)

File: react-cra/src/App.test.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { render, screen } from '@testing-library/react';
 import App from './App';
+import '@testing-library/jest-dom/extend-expect';
```
Comment (jbedard) [inline react-cra/src/App.test.tsx:3]: This wasn't required before?

**Review by jbedard — APPROVED** [review]

File: react-cra/package.json [hunk 2]
```diff
@@ -3,26 +3,34 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@testing-library/jest-dom": "5.16.5",
+    "@testing-library/jest-dom": "4.2.4",
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply react-cra/package.json:6]: Yeah. With 5.x the `@testing-library/jest-dom/extend-expect` line didn't work

File: react-cra/src/App.test.tsx (hunk 1 above)
Comment (gregmagolan) [reply react-cra/src/App.test.tsx:3]: Wasn't needed with transpiling in Jest. I think it was magically done in `setupTests.ts` with the `@testing-library/jest-dom` import but that doesn't work with separate ts_project transpiliation.

**Pull request merged**